translation-web/src/main/java/eu/europeana/api/translation/web/service/LangDetectionWebService.java

```diff
@@ -53,10 +58,38 @@ public LangDetectResponse detectLang(LangDetectRequest langDetectRequest)
         }
       }
     }
+    return new LangDetectResponse(accumulateLanguages(langDetectRequest.getText(), langs, eligibleValues), serviceId);
+  }
 
-    return new LangDetectResponse(langs, serviceId);
+  /**
+   * Returns languages detected and null responses for the non-eligible values
+   * @param texts texts sent in the request
+   * @param langDetected languages detected by the service
+   * @param eligibleValues text sent for language detection
+   * @return
+   */
+  private List<String> accumulateLanguages(List<String> texts, List<String> langDetected, List<String> eligibleValues) {
```

Comment:
In order to have consistent implementation between services I would prefer to change the Language detection to have similar implementation with the translation service, meaning to use a LanguageDetectionObj (as we already have 4 fields for that object: 1 - text, language, eligible, hint)